packages/jellyfish-api-core/__tests__/category/loan/setLoanToken.test.ts

```diff
@@ -0,0 +1,280 @@
+      symbol: 'Token6',
+      name: 'Token6',
+      priceFeedId
+    })
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].name).toStrictEqual('Token6')
+  })
+
+  it('should setLoanToken if name is more than 128 letters', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token7',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token7',
+      name: 'x'.repeat(129), // 129 letters
+      priceFeedId
+    })
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].name).toStrictEqual('x'.repeat(128)) // Only remain the first 128 letters.
+  })
+
+  it('should setLoanToken if mintable is false', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token8',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token8',
+      priceFeedId,
+      mintable: false
+    })
+    expect(typeof loanTokenId).toStrictEqual('string')
+    expect(loanTokenId.length).toStrictEqual(64)
+    await testing.generate(1)
+  })
+
+  it('should setLoanToken if interest number is greater than 0 and has less than 9 digits in the fractional part', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token9',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token9',
+      priceFeedId,
+      interest: new BigNumber(15.12345678) // 8 digits in the fractional part
+    })
+    expect(typeof loanTokenId).toStrictEqual('string')
+    expect(loanTokenId.length).toStrictEqual(64)
+    await testing.generate(1)
+  })
+
+  it('should not setLoanToken if interest number is greater than 0 and has more than 8 digits in the fractional part', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token10',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const promise = testing.rpc.loan.setLoanToken({
+      symbol: 'Token10',
+      priceFeedId,
+      interest: new BigNumber(15.123456789) // 9 digits in the fractional part
+    })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Invalid amount\', code: -3, method: setloantoken')
+  })
+
+  it('should setLoanToken if interest number is lesser than 0', async () => {
```

Comment:
Regardless number of digits in the fractional part, if less than 0 , `Amount out of range ` error will be thrown